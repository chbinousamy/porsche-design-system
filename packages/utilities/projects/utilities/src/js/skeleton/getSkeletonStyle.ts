import { borderRadiusMedium, borderRadiusSmall } from '../border';
import { themeLightBackgroundSurface } from '../theme';

type BorderRadius = 'small' | 'medium';
type Options = {
  borderRadius?: BorderRadius | string;
};

export const getSkeletonStyle = (opts?: Options) => {
  const { borderRadius = 'small' } = opts || {};
  const borderRadiusValue =
    borderRadius === 'small'
      ? borderRadiusSmall
      : borderRadius === 'medium'
      ? borderRadiusMedium
      : borderRadius || borderRadiusSmall;

  return {
    display: 'block',
    background: `${themeLightBackgroundSurface} linear-gradient(to right, transparent 0%, #f7f7f7 20%, transparent 50%) 0 0 / 200% 100%`,
    borderRadius: borderRadiusValue,
    animation:
      'skeletonAnimation var(--transition-duration-long, .6s) var(--easing-base, cubic-bezier(0.25,0.1,0.25,1)) infinite', // TODO: use motion variables
    '@keyframes skeletonAnimation': {
      from: { backgroundPosition: '100% 0' },
      to: { backgroundPosition: '-100% 0' },
    },
  } as const;
};
