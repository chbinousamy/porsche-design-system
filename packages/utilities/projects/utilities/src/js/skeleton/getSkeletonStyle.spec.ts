import { getSkeletonStyle } from './getSkeletonStyle';
import * as fromSkeleton from './';

it('should provide all exports', () => {
  expect(Object.keys(fromSkeleton).length).toBe(1);
});

describe('getSkeletonStyle()', () => {
  it.each<Parameters<typeof getSkeletonStyle>>([
    [{ borderRadius: undefined, theme: 'light' }],
    [{ borderRadius: 'small', theme: 'light' }],
    [{ borderRadius: 'medium', theme: 'light' }],
    [{ borderRadius: '6px', theme: 'light' }],
    [{ borderRadius: 'small', theme: 'dark' }],
    [{ borderRadius: 'medium', theme: 'dark' }],
  ])('should return correct css for opts: %s', (...args) => {
    expect(getSkeletonStyle(...args)).toMatchSnapshot();
  });
});
