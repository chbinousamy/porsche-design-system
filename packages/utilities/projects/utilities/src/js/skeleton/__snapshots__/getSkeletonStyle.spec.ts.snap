// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSkeletonStyle() should return correct css for opts: { borderRadius: '6px', theme: 'light' } 1`] = `
{
  "@keyframes skeletonAnimation": {
    "from": {
      "backgroundPositionX": "100%",
    },
    "to": {
      "backgroundPositionX": "-100%",
    },
  },
  "animation": "skeletonAnimation var(--transition-duration-long, .6s) var(--easing-base, cubic-bezier(0.25,0.1,0.25,1)) infinite",
  "background": "#EEEFF2 linear-gradient(to right, transparent 0%, #f7f7f7 25%, transparent 50%) 0 0 / 200% 100%",
  "borderRadius": "6px",
  "display": "block",
}
`;

exports[`getSkeletonStyle() should return correct css for opts: { borderRadius: 'medium', theme: 'dark' } 1`] = `
{
  "@keyframes skeletonAnimation": {
    "from": {
      "backgroundPositionX": "100%",
    },
    "to": {
      "backgroundPositionX": "-100%",
    },
  },
  "animation": "skeletonAnimation var(--transition-duration-long, .6s) var(--easing-base, cubic-bezier(0.25,0.1,0.25,1)) infinite",
  "background": "#212225 linear-gradient(to right, transparent 0%, #1a1b1e 25%, transparent 50%) 0 0 / 200% 100%",
  "borderRadius": "8px",
  "display": "block",
}
`;

exports[`getSkeletonStyle() should return correct css for opts: { borderRadius: 'medium', theme: 'light' } 1`] = `
{
  "@keyframes skeletonAnimation": {
    "from": {
      "backgroundPositionX": "100%",
    },
    "to": {
      "backgroundPositionX": "-100%",
    },
  },
  "animation": "skeletonAnimation var(--transition-duration-long, .6s) var(--easing-base, cubic-bezier(0.25,0.1,0.25,1)) infinite",
  "background": "#EEEFF2 linear-gradient(to right, transparent 0%, #f7f7f7 25%, transparent 50%) 0 0 / 200% 100%",
  "borderRadius": "8px",
  "display": "block",
}
`;

exports[`getSkeletonStyle() should return correct css for opts: { borderRadius: 'small', theme: 'dark' } 1`] = `
{
  "@keyframes skeletonAnimation": {
    "from": {
      "backgroundPositionX": "100%",
    },
    "to": {
      "backgroundPositionX": "-100%",
    },
  },
  "animation": "skeletonAnimation var(--transition-duration-long, .6s) var(--easing-base, cubic-bezier(0.25,0.1,0.25,1)) infinite",
  "background": "#212225 linear-gradient(to right, transparent 0%, #1a1b1e 25%, transparent 50%) 0 0 / 200% 100%",
  "borderRadius": "4px",
  "display": "block",
}
`;

exports[`getSkeletonStyle() should return correct css for opts: { borderRadius: 'small', theme: 'light' } 1`] = `
{
  "@keyframes skeletonAnimation": {
    "from": {
      "backgroundPositionX": "100%",
    },
    "to": {
      "backgroundPositionX": "-100%",
    },
  },
  "animation": "skeletonAnimation var(--transition-duration-long, .6s) var(--easing-base, cubic-bezier(0.25,0.1,0.25,1)) infinite",
  "background": "#EEEFF2 linear-gradient(to right, transparent 0%, #f7f7f7 25%, transparent 50%) 0 0 / 200% 100%",
  "borderRadius": "4px",
  "display": "block",
}
`;

exports[`getSkeletonStyle() should return correct css for opts: { borderRadius: undefined, theme: 'light' } 1`] = `
{
  "@keyframes skeletonAnimation": {
    "from": {
      "backgroundPositionX": "100%",
    },
    "to": {
      "backgroundPositionX": "-100%",
    },
  },
  "animation": "skeletonAnimation var(--transition-duration-long, .6s) var(--easing-base, cubic-bezier(0.25,0.1,0.25,1)) infinite",
  "background": "#EEEFF2 linear-gradient(to right, transparent 0%, #f7f7f7 25%, transparent 50%) 0 0 / 200% 100%",
  "borderRadius": "4px",
  "display": "block",
}
`;
